### ISSUE TITLE
Parameter groups

### ISSUE THREAD

**Issue by vladimirjovanovicTT** [tenstorrent/tt-forge-fe#1693]
We want to be able to set different hyperparameters (e.g. learning rate, optimizer alpha, beta) for different parameters/parameter groups (proven to improve results in practice in various experiments).
